src/source_sampling.cpp

```diff
@@ -92,7 +94,10 @@ std::vector<double> pyne::Sampler::particle_birth(std::vector<double> rands) {
   samp.push_back(pos[2]); 
   samp.push_back(sample_e(e_idx, rands[5]));
   samp.push_back(sample_w(pdf_idx));
-  return samp;
+  samp.push_back(-1.0);
```

Comment:
There is no need for the vector `samp` anymore... just initialize `SourceParticle` directly from `pos`, etc